### PR TITLE
feat: Declarative-friendly resources should avoid custom methods.

### LINKED ISSUE #672: AIP-136: Declarative-friendly resources should avoid custom methods.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Fixes #672.

### CODE REVIEW

File: rules/aip0136/declarative_standard_methods_only.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var standardMethodsOnly = &lint.MethodRule{
+	Name:   lint.NewRuleName(136, "declarative-standard-methods-only"),
+	OnlyIf: utils.IsDeclarativeFriendlyMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Standard methods are fine.
+		standard := stringset.New("Get", "List", "Create", "Update", "Delete", "Undelete")
```
Comment (apasel422) [inline rules/aip0136/declarative_standard_methods_only.go:31]: Should the batch methods be included here?

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: rules/aip0136/declarative_standard_methods_only.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/aip0136/declarative_standard_methods_only.go:31]: Oh, probably.

Comment (lukesneeringer) [reply rules/aip0136/declarative_standard_methods_only.go]: Done.

**Review by apasel422 — APPROVED** [review]

File: rules/aip0136/declarative_standard_methods_only_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		MethodName string
+		ImpOnly    string
+		problems   testutils.Problems
+	}{
+		{"ValidGet", "GetBook", "", nil},
+		{"ValidList", "ListBooks", "", nil},
+		{"ValidCreate", "CreateBook", "", nil},
+		{"ValidUpdate", "UpdateBook", "", nil},
+		{"ValidDelete", "DeleteBook", "", nil},
+		{"ValidUndelete", "UndeleteBook", "", nil},
```
Comment (apasel422) [inline rules/aip0136/declarative_standard_methods_only_test.go:35]:
```suggestion
		{"ValidUndelete", "UndeleteBook", "", nil},
		{"ValidBatch", "BatchGetBooks", "", nil},
```

**Pull request merged**